File: Health/HealthCare.Api/Models/Appointment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HealthCare.Api.Models
{
    public class Appointment
    {
        [Key]
        public int AppointmentId { get; set; }
        public int PatientId { get; set; }
        public int DoctorId { get; set; }

        [Required]
        public DateOnly ScheduledDate { get; set; }

        [Required]
        [MaxLength(20)]
        public string TimeSlot { get; set; } = null!;

        [Required]
        [MaxLength(20)]
        public string Status { get; set; } = "Pending";  

        [MaxLength(500)]
        public string? CancellationReason { get; set; }

        public DateTimeOffset CreatedDate { get; set; } = DateTimeOffset.UtcNow;

        // Navigation
        [ForeignKey(nameof(PatientId))]
        public Patient Patient { get; set; } = null!;

        [ForeignKey(nameof(DoctorId))]
        public Doctor Doctor { get; set; } = null!;

        public HealthRecord? HealthRecord { get; set; }
    }
}

File: Health/HealthCareUser.Web/src/app/cores/models/Appointment.ts
export interface Appointment {

  appointmentId: number;

  patientId: number;
  patientName: string;

  doctorId: number;
  doctorName: string;

  scheduledDate: string;
  timeSlot: string;

  status: string;

}

export interface DoctorDropdownDto {
  doctorId: number;
  fullName: string;
  specialization: string;
}

export interface CreateAppointmentDto {
  doctorId: number;
  scheduledDate: string;
  timeSlot: string;
  reason?: string;
}
